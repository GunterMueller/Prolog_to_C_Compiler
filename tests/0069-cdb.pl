main :-
	asserta((foo :- bar)),
	asserta((foo :- baz)),
	assertz(hello(X)),
	assertz(hello(1)),
	asserta(hello(no, ok), REF),
	clause(H, B, REF), H == hello(no, ok), B == true,
	retract(hello(X)), display(X), nl,
	show(foo),
	nl, show(hello(_)),
	abolish([foo/0, hello/1]),
	nl, show(foo),
	nl, show(hello(_, _)),
	erase(REF),
	nl, show(hello(_, _)).

show(HEAD) :-
	clause(HEAD, BODY), write((HEAD :- BODY)), nl, fail.
show(_).
