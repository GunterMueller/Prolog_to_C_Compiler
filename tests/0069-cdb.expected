_15
foo :- baz
foo :- bar

hello(1) :- true


hello(no,ok) :- true

